mkosi/config.py

```diff
@@ -44,13 +47,43 @@ def parse_boolean(s: str) -> bool:
     die(f"Invalid boolean literal: {s!r}")
 
 
+def parse_path(value: str, required: bool, absolute: bool = True, expanduser: bool = True, expandvars: bool = True) -> Path:
+    if expandvars:
+        path = Path(string.Template(value).substitute(os.environ))
+    else:
+        path = Path(value)
+
+    if expanduser:
+        user = current_user()
+        if (
+            not path.is_absolute()
+            and path.parts
+            and path.parts[0] == "~"
+            and os.getuid() != user.pw_uid
+        ):
+            path = Path(user.pw_dir, *path.parts[1:])
```

Comment:
```suggestion
            path = Path(user.pw_dir) / path.relative_to("~")
```
though I'd find the expanding `Path("~username")` and calling `Path.expanduser()` unconditionally clearer still.

Reply:
I used your suggestion.

However I'm not sure I agree with `Path.expanduser()` being clearer, we are clearly doing something special (and we already have access to the home directory).